pkg/sql/sem/tree/function_definition.go

```diff
@@ -28,6 +28,30 @@ type FunctionDefinition struct {
 	FunctionProperties
 }
 
+// Volatility indicates whether the result of a function is dependent *only*
+// on the values of its explicit arguments, or can change due to outside factors
+// (such as parameter variables or table contents).
+//
+// Each value here must be ordered such that the property i+1 is "weaker" the property
+// i, e.g. "Stable" volatilities also work for "Immutable" volatilities.
+//
+// NOTE: functions having side-effects, such as setval(),
+// must be labeled volatile to ensure they will not get optimized away,
+// even if the actual return value is not changeable.
+type Volatility int
+
+const (
+	// VolatilityImmutable indicates the builtin result never changes for a
+	// given input.
+	VolatilityImmutable Volatility = iota
```

Comment:
I think we should use the characters here. That eases the stringification and also makes the zero value something that should never happen (and thus always incorrect if it does).

Reply:
fine with me.